### ISSUE TITLE
error 500 on windows trying to use ruby

### ISSUE THREAD

**Issue by rismoney** [jung-kurt/caddy-cgi#5]
caddyfile
```shell
http://:80 {
  log stdout
  root C:\wwwroot
  cgi /*.rb C:\ruby\bin\ruby.exe C:\cgi\cgi.rb
}
```

works fine with html in C:\wwwroot

contents of cgi.rb (located in C:\cgi):
```ruby
require 'cgi'
cgi = CGI.new

msg = 'hello'

begin
  puts cgi.header('type' => 'text/plain')
  puts msg
rescue => e
  STDERR.puts e.message
end
```

file works fine interactively and on other cgi web services elsewhere

**Comment by jung-kurt** [comment]
Usually these problems have to do with missing environment variables. Is RUBYLIB set in your command line environment? If so, you may need to pass it through Caddy with the pass_env subdirective. Alternatively, try `require '/path/to/cgi`, but there may be trouble if cgi itself needs to require a library.

**Comment by rismoney** [comment]
interesting, going to try your suggestion- 

in my caddy logs i see this:
```2017/09/05 18:07:46 http://
2017/09/05 18:08:01 cgi: bogus header line:
2017/09/05 18:08:01 cgi: bogus header line: hello
2017/09/05 18:08:01 cgi: no headers
```

so it's definitely running ruby, but maybe not handling the cgi gem properly (which i believe to be a built-in)

**Comment by magikstm** [comment]
I'm unsure as I don't use ruby.

But, could it be that you're missing this first line in your cgi.rb file?
#!/path/to/ruby ruby

Kinda similar to linux:
#!/usr/bin/env ruby

Ref:
http://editrocket.com/articles/ruby_apache_windows.html
http://rubychallenger.blogspot.ca/2011/04/how-to-run-ruby-cgi-scripts-in-windows.html
https://stackoverflow.com/questions/2245634/setting-up-ruby-cgi-in-apache

That line isn't listed above in your cgi.rb file.

**Comment by jung-kurt** [comment]
> But, could it be that you're missing this first line in your cgi.rb file?

> `#!/path/to/ruby ruby`

CGI for Caddy can launch a script with a shebang like this, or it can specify the interpreter right in the `cgi` directive. In this case, the latter method is used so no shebang line is required.

**Comment by rismoney** [comment]
After much trial and tribulation, I determined, that cgi.header was generating headers suffixed with `\r\n\r\n` aka CRLF.   This apparently breaks caddy's header processing, so for me I did a :

`print cgi.header('type' => 'text/plain').encode(universal_newline: true)`

Note print does not modify line endings, so we want to use print.
After the header is set, everything works great.

HTH someone who reads this.
